src/linter-jscs.js

```diff
@@ -105,7 +108,11 @@ export default class LinterJSCS {
             this.fixString(editor);
           }
         }
-      });
+      }));
+      currentEditor.add(editor.onDidDestroy(() => {
+        currentEditor.dispose();
```

Comment:
What I gather from the docs and the referenced solution is that this `currentEditor` represents a group of event handlers, for the argument `editor`. So when the `onDidDestroy` happens we need to dispose this entire group.
Otherwise we'd probably have a memory leak in the Map, which is required as well.
I'll try to clarify this a bit